src/nupic/algorithms/AnomalyLikelihood.cpp

```diff
@@ -0,0 +1,158 @@
+#include <iostream>
+#include <vector>
+#include <unordered_map>
+#include <string>
+
+#include <nupic/types/Types.hpp>
+
+using namespace std;
+
+struct DistributionParams {
+  DistributionParams(string name, Real32 mean, Real32 variance, Real32 stdev) :
+    name(name),mean(mean), variance(variance), stdev(stdev) {}
+  string name;
+  Real32 mean;
+  Real32 variance;
+  Real32 stdev;
+};
+
+struct DataPoint {
+    DataPoint(string value, double anomalyScore, int timestamp, bool timestampPassed) :
+    value(value),
+    anomalyScore(anomalyScore),
+    timestamp(timestamp),
+    timestampSeted(timestampPassed) { }
+    string value;
+    double anomalyScore;
+    int timestamp;
+    bool timestampSeted;
+};
+
+class MovingAverage {
```

Comment:
no need for this. Remove and please use the impl from `nupic/utils/MovingAverage`